src/containers/ChangeButtonContainer.js

```diff
@@ -0,0 +1,20 @@
+import { connect } from 'react-redux';
+import { toggleMode } from '../actions/actions';
+import { ChangeToTagButton } from '../views/ChangeModeButton';
+
+function mapStateToProps(state) {
+  return {
+    isTagMode: state.isTagMode,
+  };
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    changeGameMode: () => dispatch(toggleMode()),
+  };
```

Comment:
for the simple cases (like here, or if the closure passes the exact same arguments to the action creator)  of `mapDispatchToProps`, [it's recommended to use the object notation instead of the function notation](https://react-redux.js.org/using-react-redux/connect-mapdispatch#two-forms-of-mapdispatchtoprops). 

helps in a couple ways:
- cleaner and easier to read
- reduces the testing surface area
    - a lot of coverage tools would flag this function as "untested" even though it's rather unnecessary for developers in most cases since it's testing redux internals

so we would convert this to:

```js
const mapDispatchToProps = {
  changeGameMode: toggleMode
};
```

Reply:
Since we aren't calling dispatch like below, so should mapDispatchToProps just be removed in situations where we pass object notation instead of function notation? 

I guess I don't understand what's going on in the react-redux's connect function. I thought I had to pass the action I wanted to dispatch and pass that higher order function as a prop to the component I want it in.

Reply:
if we're using `dispatch`, yes we'll need a `mapDispatchToProps` so that we can properly get `dispatch` to wrap our actions. object notation is valid because of how the `connect` internals work. If it sees the second argument (mapDispatchToProps) is an object, it will wrap all object values (the actions) with `dispatch`. conceptually it'll automatically do the same thing as we've done manually here.

for the case below (https://github.com/rayzlui/waldo/pull/9#discussion_r378323999), in that `mapDispatchToProps`, we're not using `dispatch` at all, so it isn't necessary. we could let `selectImage` be passed through this container layer and use it directly in the component. ie:

```js
// Some consumption file
<MyContainer selectImage={() => // do stuff} />

...

// Container file
// Note, we don't do anything with selectImage here
export const MyContainer = connect(mapStateToProps)(MyComponent);

...

// View file
export function MyComponent(props) {
  const { selectImage } = props;

   // use selectImage directly in the component here
}

```